CMakeLists.txt

```diff
@@ -1154,81 +1138,25 @@ endif()
 
 if(APPLE AND NOT IOS)
     message(STATUS "Building on OSX")
-    add_definitions(-DMACOSX -DPIPES -DNO_FLTK_THREADS -DHAVE_SOCKETS)
+    add_compile_definitions("MACOSX" "PIPES" "NO_FLTK_THREADS" "HAVE_SOCKETS")
     find_library(ACCELERATE_LIBRARY Accelerate)
     find_path(VECLIB_PATH "Accelerate/Accelerate.h")
     list(APPEND libcsound_private_include_dirs ${VECLIB_PATH})
     list(APPEND libcsound_LIBS ${MATH_LIBRARY} dl ${ACCELERATE_LIBRARY})
 endif()
 
 if(WIN32)
-    add_definitions(-DWIN32)
-endif()
-
-check_function_exists(strlcat HAVE_STRLCAT)
-if(HAVE_STRLCAT)
-    add_definitions(-DHAVE_STRLCAT)
-endif()
-
-# Locale-aware reading and printing
-check_function_exists(strtok_r HAVE_STRTOK_R)
-check_function_exists(strtod_l HAVE_STRTOD_L)
-check_function_exists(sprintf_l HAVE_SPRINTF_L)
-
-if(HAVE_STRTOK_R)
-    add_definitions(-DHAVE_STRTOK_R)
-endif()
-if(HAVE_STRTOD_L)
-    add_definitions(-DHAVE_STRTOD_L)
-endif()
-if(HAVE_SPRINTF_L)
-    add_definitions(-DHAVE_SPRINTF_L)
-endif()
-
-# Same for Windows
-check_function_exists(_strtok_r HAVE__STRTOK_R)
-check_function_exists(_strtod_l HAVE__STRTOD_L)
-check_function_exists(_sprintf_l HAVE__SPRINTF_L)
-
-if(HAVE__STRTOK_R)
-    add_definitions(-DHAVE__STRTOK_R)
-endif()
-if(HAVE__STRTOD_L)
-    add_definitions(-DHAVE__STRTOD_L)
-endif()
-if(HAVE__SPRINTF_L)
-    add_definitions(-DHAVE__SPRINTF_L)
+    add_compile_definitions("WIN32")
 endif()
 
 # Pass flags according to system capabilities
 
-if(HAVE_WINSOCK_H OR HAVE_SYS_SOCKETS_H)
+if(HAVE_WINSOCK_H OR HAVE_SYS_SOCKET_H)
     list(APPEND libcsound_CFLAGS -DHAVE_SOCKETS)
 endif()
-if(HAVE_DIRENT_H)
-    list(APPEND libcsound_CFLAGS -DHAVE_DIRENT_H)
-endif()
-if(HAVE_FCNTL_H)
-    list(APPEND libcsound_CFLAGS -DHAVE_FCNTL_H)
-endif()
-if(HAVE_UNISTD_H)
-    list(APPEND libcsound_CFLAGS -DHAVE_UNISTD_H)
-endif()
 if(HAVE_STDINT_H)
     list(APPEND libcsound_CFLAGS -DHAVE_STDINT_H)
 endif()
-if(HAVE_SYS_TIME_H)
-    list(APPEND libcsound_CFLAGS -DHAVE_SYS_TIME_H)
-endif()
-if(HAVE_SYS_TYPES_H)
-    list(APPEND libcsound_CFLAGS -DHAVE_SYS_TYPES_H)
-endif()
-if(HAVE_TERMIOS_H)
-    list(APPEND libcsound_CFLAGS -DHAVE_TERMIOS_H)
-endif()
-if(HAVE_VALUES_H)
-    list(APPEND libcsound_CFLAGS -DHAVE_VALUES_H)
-endif()
```

Comment:
Where have these moved to?

Reply:
Did these in batch with `HEADERS_TO_CHECK`. But will probably have to do them individually if we want to specify which target to add them too.

Reply:
Decided not to change this